src/codegeneration/JsxTransformer.js

```diff
@@ -0,0 +1,163 @@
+// Copyright 2015 Traceur Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the 'License');
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an 'AS IS' BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  JSX_ELEMENT,
+  JSX_PLACEHOLDER,
+  JSX_TEXT,
+} from '../syntax/trees/ParseTreeType.js';
+import {
+  JsxText,
+  LiteralExpression,
+  LiteralPropertyName,
+} from '../syntax/trees/ParseTrees.js';
+import {ParseTreeTransformer} from './ParseTreeTransformer.js';
+import {STRING} from '../syntax/TokenType.js';
+import {
+  createArgumentList,
+  createIdentifierToken,
+  createMemberExpression,
+  createNullLiteral,
+  createObjectLiteral,
+  createPropertyNameAssignment,
+  createStringLiteral,
+  createStringLiteralToken,
+} from './ParseTreeFactory.js';
+import {parseExpression} from './PlaceholderParser.js';
+
+/**
+ * Desugars JSX expressions.
+ *
+ *   <p a="b" c="d">e{f}g</p>
+ *
+ * to:
+ *
+ *   Rect.createElement('p', {a: 'b', c: 'd'}, 'e', f, 'g')
+ *
+ * The function to use is be default React.createElement but it is configurable.
```

Comment:
I guess you meant: 
The constructor emitted is configurable; it defaults to Rect.createElement().